### ISSUE TITLE
Installation issues - CUDA

### ISSUE THREAD

**Issue by Joeetaku** [facebook/fbcunn#68]
Hi all,

I'm having trouble installing fbcunn according to the instructions at https://github.com/facebook/fbcunn/blob/master/INSTALL.md

I'm stuck at the final instruction, as I apparently have the incorrect version of CUDA installed, even though I'm sure it's the latest version.
 
--------------------------------------------

    $ luarocks make rocks/fbcunn-scm-1.rockspec
       git submodule init
       git submodule update
    cmake -E make_directory build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/home/jtownsend/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/home/jtownsend/torch/install/lib/luarocks/rocks/fbcunn/scm-1"

    -- Found Torch7 in /home/jtownsend/torch/install
    CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
      Could NOT find CUDA: Found unsuitable version "6.0", but required is at
      least "6.5" (found /usr)
    Call Stack (most recent call first):
      /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:313 (_FPHSA_FAILURE_MESSAGE)
      /usr/share/cmake-2.8/Modules/FindCUDA.cmake:806 (find_package_handle_standard_args)
      CMakeLists.txt:13 (FIND_PACKAGE)

    -- Configuring incomplete, errors occurred!
    See also "/home/jtownsend/fbcunn/build/CMakeFiles/CMakeOutput.log".

    Error: Build error: Failed building.

--------------------------------------------

Running ./deviceQuery tells me that my CUDA driver and runtime versions are both 7.0. 'nvcc --version' also tells me that the compiler driver is version 7.0. I have followed post-installation instructions (http://docs.nvidia.com/cuda/cuda-getting-started-guide-for-linux/index.html#post-installation-actions) to set environment variables such that they point to '/usr/local/cuda-7.0'.

But still, I receive the errors above.

Any ideas?

Kind regards,

Joe

**Comment by soumith** [comment]
is it possible that you have two CUDA versions on your machine? does /usr/local/cuda point to 6.0?

**Comment by Joeetaku** [comment]
No, it definitely points to 7.0. Furthermore usr/local/ only contains cuda-7.0, no cuda-6.0.

I had considered this possibility and I'm working on the assumption that it's something along such lines... but I can't find any trace of a cuda-6.0 installation anywhere on my system!

**Comment by soumith** [comment]
that's super weird. it finds the CUDA stuff in /usr, your cuda install is in /usr/local

**Comment by Joeetaku** [comment]
Yeah, I thought that strange too, especially as /usr does not contain anything cuda related in that immediate directory. 

However there are nvidia folders in other subdirectories. I just noticed that /usr/lib32/ contains 'nvidia-346', and /usr/lib/ contains this plus 'nvidia', 'nvidia-346-prime',  'nvidia-cuda-toolkit' and 'nvidia-visual-profiler'.... I don't suppose there's anything I should be looking for in any of these?

Alternatively, is there any way I can force the fbcunn script to look for CUDA in usr/local/cuda-7.0?

Forgive me if I don't reply for a while... I'll be away until Monday!

**Comment by Joeetaku** [comment]
Clearing the cmake cache in fbcunn/build/CMakeCache.txt appears to have resolved this.

I am experiencing other problems now, but they appear to be completely unrelated. Will make a separate topic for those if I'm still scratching my head later on!

**Comment by mszlazak** [comment]
Thanks. Will a purge on "fbcunn*" remove all of the Facebook extras? Maybe putting up a note on the fbcunn install page on un-installing it would be nice.
